infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/RestApiRequest.java

```diff
@@ -84,17 +86,36 @@ public void respondWithCode(final int statusCode) {
     context.status(statusCode);
   }
 
-  public String getPathParam(final String pathParameter) {
-    return pathParamMap.get(pathParameter);
+  public <T> T getPathParameter(final ParameterMetadata<T> parameterMetadata)
+      throws BadRequestException {
+    try {
+      return parameterMetadata
+          .getType()
+          .deserializeFromString(pathParamMap.get(parameterMetadata.getName()));
+    } catch (IOException e) {
+      throw new BadRequestException("Failed to get path parameter " + e.getMessage());
+    }
   }
 
-  public int getQueryParamAsInt(final String queryParameter) {
-    return SingleQueryParameterUtils.getParameterValueAsInt(queryParamMap, queryParameter);
+  public <T> Optional<T> getOptionalQueryParameter(final ParameterMetadata<T> parameterMetadata)
+      throws IOException {
```

Comment:
I don't *think* we need to throw `IOException` here now that we're not actually using the JSON parser.